src/views/TabView/TabView.js

```diff
@@ -110,54 +110,56 @@ class TabView extends PureComponent<void, Props, void> {
 
   _renderTabBar = (props: *) => {
     const {
-      tabBarOptions,
-      tabBarComponent: TabBarComponent,
       animationEnabled,
+      navigation,
+      router,
+      screenProps,
+      tabBarComponent: TabBarComponent,
+      tabBarOptions,
     } = this.props;
     if (typeof TabBarComponent === 'undefined') {
       return null;
     }
+
+    const { state } = navigation;
+    const options = router.getScreenOptions(
+      this.props.childNavigationProps[state.routes[state.index].key],
+      screenProps || {}
+    );
+
+    const tabBarVisible = options.tabBarVisible === null
+      ? true
+      : options.tabBarVisible;
+
     return (
       <TabBarComponent
         {...props}
         {...tabBarOptions}
-        screenProps={this.props.screenProps}
         navigation={this.props.navigation}
         getLabel={this._getLabel}
         renderIcon={this._renderIcon}
         animationEnabled={animationEnabled}
+        style={!tabBarVisible && styles.hidden }
```

Comment:
This is unclear syntax for many people, please use the more conventional `style={!tabBarVisible ? styles.hidden : undefined}`